llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -2226,6 +2226,10 @@ class TargetInstrInfo : public MCInstrInfo {
     return OptLevel >= CodeGenOptLevel::Aggressive ? 4 : 2;
   }
 
+  /// Returns the target-specific default value for tail merging.
+  /// This value will be used if the tail-merge-size argument is not provided.
+  virtual unsigned getTailMergeSize() const { return 3; }
```

Comment:
Target hooks without any arguments are generally bad. Should make a static property, or pass in the function or something